### PR TITLE
Faq feature

### PR DESCRIPTION
**Pull request by karki93pankaj**
Add and Edit Faq with basic fields functionality working

### CODE REVIEW

File: frontend/src/components/Faqs/Add/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import PropTypes from 'prop-types'
+import gql from 'graphql-tag'
+
+import Layout from '../../Layout'
+import PublishButton from '../../Generic/PublishButton'
+import FaqForm from '../FaqForm';
+import { SAVE_FAQ_TO_DB } from '../mutations'
+import { GET_FAQ } from '../queries'
+
+const CURRENT_USER = gql`
```
Comment (sumandomarpa) [inline frontend/src/components/Faqs/Add/index.js]: Query should be saved in the queries.js file

File: frontend/src/components/Faqs/Add/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    me @client {
+      id
+    }
+  }
+`
+
+class AddFaq extends Component {
+  handleSubmit = async () => {
+    const { client, history } = this.props
+    const { faq } = client.readQuery({ query: GET_FAQ })
```
Comment (sumandomarpa) [inline frontend/src/components/Faqs/Add/index.js]: For `Faq` we don't want to store it to the apollo local cache. The component level state should be fine I suppose.
We are only using local cache for the `Page` module, as it has complicated logic.

File: frontend/src/components/Faqs/Edit/index.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  }
+`
+
+class EditFaq extends Component {
+  state = {
+    isDataLoaded : false
+  }
+  componentDidMount () {
+    this.fetchFromDBtoCache()
```
Comment (sumandomarpa) [inline frontend/src/components/Faqs/Edit/index.js]: No cache, as mentioned above.

File: frontend/src/components/Faqs/Edit/index.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import gql from 'graphql-tag'
+
+import Layout from '../../Layout'
+import PublishButton from '../../Generic/PublishButton'
+import FaqForm from '../FaqForm';
+import { UPDATE_FAQ_TO_DB } from '../mutations'
+import { GET_FAQ , GET_FAQ_DB} from '../queries'
+
+const CURRENT_USER = gql`
+  query {
```
Comment (sumandomarpa) [inline frontend/src/components/Faqs/Edit/index.js]: Query should be in queries.js, as mentioned above.

File: frontend/src/components/Faqs/styles.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import styled from 'styled-components'
+
+const AddNewFaqWrapper = styled.div`
```
Comment (sumandomarpa) [inline frontend/src/components/Faqs/styles.js]: If this one is not used, maybe we can get rid of it.

File: frontend/src/index.js [hunk 6]
```diff
@@ -51,6 +52,16 @@ cache.writeData({
     ],
     boxes: [],
     prosAndCons: [],
+    faq: {
```
Comment (sumandomarpa) [inline frontend/src/index.js]: We should not use local cache, as mentioned above.

File: frontend/src/resolvers.js [hunk 7]
```diff
@@ -278,5 +278,18 @@ export const resolvers = {
       if (prosAndConsDoc.cons.length > 1)
         remove(prosAndConsDoc.cons, cons => cons.id === prosOrConsId)
     },
+    updateFaq: (_root, variables, { cache, getCacheKey }) => {
+      const { name, value, faqId } = variables
```
Comment (sumandomarpa) [inline frontend/src/resolvers.js]: After removing local cache, this one should be removed as well.

**Comment by karki93pankaj** [comment]
@sumandomarpa - I have incorporated all the asked changes. Please let me know if any other changes are required.

File: frontend/src/components/Faqs/FaqForm.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        <InputBox name="title" value={title} label="Title"
+          onChange={this.handleInputChange} />
+        <EditorBox name="description" value={description} label="Description"
+          onChange={this.handleInputChange} />
+        <InputBox type="textarea" name="short_description" value={short_description} 
+          label="Short Description" onChange={this.handleInputChange} />
+        <InputBox name="slug" value={slug} label="Slug"
+          placeholder="Slug" onChange={this.handleInputChange} />
+        <SelectBox name="vertical" value={vertical} label="Vertical"
+          onChange={this.handleInputChange} options={VERTICAL_OPTIONS} />
```
Comment (sumandomarpa) [inline frontend/src/components/Faqs/FaqForm.js]: In `Faq/Add` component, we are not passing any initial values right? 
How will this work if the user don't change the value of the select box?

The value in state would be empty string, and in GUI the user will see `home-loans`?

**Pull request merged**